test/e2e/middleware-rewrites/test/index.test.ts

```diff
@@ -151,6 +151,33 @@ describe('Middleware Rewrite', () => {
       expect(await browser.eval('next.router.asPath')).toBe('/param-1')
     })
 
+    it('should trigger getServerSideProps on back navigation when middleware is present', async () => {
```

Comment:
This test case seems to pass fine without the changes in this PR, are we missing the failure case?